src/dv360.ts

```diff
@@ -190,4 +191,84 @@ export class DV360 extends ApiClient {
     }
     return campaigns;
   }
+
+  protected createSdfDownloadOperation(advertiserId: string) {
+    const downloadOperation = this.fetchEntity(
+      this.getUrl('sdfdownloadtasks'),
+      {
+        method: 'post'
+      },
+      {
+        version: 'SDF_VERSION_5_5',
+        advertiserId: advertiserId,
+        parentEntityFilter: {
+          fileType: [
+            'FILE_TYPE_CAMPAIGN',
+            'FILE_TYPE_INSERTION_ORDER',
+            'FILE_TYPE_LINE_ITEM',
+            'FILE_TYPE_AD_GROUP',
+            'FILE_TYPE_AD'
+          ],
+          filterType: 'FILTER_TYPE_NONE',
+          filterIds: []
+        }
+      }
+    );
+    return downloadOperation;
+  }
+
+  protected waitForDownloadOperationResource(downloadOperationName: string) {
+    const operationResourceUrl = this.getUrl(downloadOperationName);
+    const initialDelay = 2000;
+    const maxRetries = 10;
+    const delayMultiplier = 2;
+
+    let downloadOperation;
+    let delay = initialDelay;
+    let tryCount = 0;
+    while (tryCount <= maxRetries) {
+      downloadOperation = this.fetchEntity(operationResourceUrl);
+      Logger.log(downloadOperation);
+      if (downloadOperation.done) {
+        return downloadOperation.response.resourceName;
+      }
+      Logger.log(`Backing off for ${delay}ms`);
+      Utilities.sleep(delay);
+      tryCount++;
+      delay *= delayMultiplier;
+    }
+    Logger.log(`Try limit exceeded after ${tryCount} tries`);
+    return undefined;
```

Comment:
I would "throw Error(`Try limit exceeded after ${tryCount} tries`);"
Because later you can catch those errors (and other errors from the "fetchResponse").